text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,196 @@
+---
+rfc pr: [#347](https://github.com/aws/aws-cdk-rfcs/pull/347)
+tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/66
+---
+
+# CDK Self Managed Stack Set Support
+
+As a CDK user, I would like to define stack set infrastructure through native CDK constructs,
+ and create/update self-managed stack sets through CDK deploy command.
+
+* **Original Author(s):** @linsona
+* **Tracking Issue**: #66
+* **API Bar Raiser**: @skinny85
+
+## Working Backwards
+
+### CHANGELOG
+
+* feat(assembly): add stack set artifact type and schema
+* feat(core): add stack set core construct
+* feat(cli): add logic for cli commands such as `cdk deploy` to collect stack set artifact types
+* feat(bootstrap): add default stack set roles for CDK deployments
+
+### README
+
+The `cdk.StackSet` construct in CDK defines AWS resources in a stack set and stack set
+ configuration/deployment preferences.
+ This is different from the `cdk.CfnStackSet` construct, which requires a CloudFormation stack that then deploys a CloudFormation stack set.
+ The `cdk.StackSet` construct creates the stack set directly if it does not exist,
+or updates a stack set and existing stack set instances while monitoring the stack set operation.
+
+#### Prerequisites
+
+To enable the use of self managed stack sets two roles must exist:
+
+* **Stack Set Administration Role:** Role in the parent account, which is used by CloudFormation to assume the stack set execution role in the child account.
+* **Stack Set Execution Role:** Role in the child accounts, which allows the stack set administration role in parent account to assume,
+and perform the stack set instance deployment.
+* CDK bootstrap provides default stack set roles,
+otherwise customers can manually create these roles and specify them explicitly them while defining CDK stack sets
+
+Using CDK Bootstrap to provision default cdk stack set roles:
+
+* Parent Account:
+  * Using the parent account credentials and region, run `cdk bootstrap`
+  * Specific stack sets resources in bootstrap:
+    * A staging bucket to store templates larger than 50kb
+    * A default stack set administration role
+      * `cdk-${Qualifier}-stack-set-admin-role-{AWS::AccountId}-${AWS::Region}`
+      * Ex. cdk-hnb659fds-stack-set-admin-role-111111111111-us-east-1
```

Comment:
Let's skip these details (remember, this is the customer-facing documentation - you don't have to explain all of the implementation details, you just need to provide instructions on how to use it).

Reply:
Removed -- added appendix section with the bootstrap resource additions